web/src/lib/components/onboarding-page/onboarding-theme.svelte

```diff
@@ -0,0 +1,71 @@
+<script lang="ts">
+  import { mdiArrowRight, mdiWhiteBalanceSunny, mdiMoonWaningCrescent } from '@mdi/js';
+  import Button from '../elements/buttons/button.svelte';
+  import Icon from '../elements/icon.svelte';
+  import OnboardingCard from './onboarding-card.svelte';
+  import { createEventDispatcher } from 'svelte';
+  import { colorTheme } from '$lib/stores/preferences.store';
+  import { browser } from '$app/environment';
+
+  const dispatch = createEventDispatcher<{
+    done: void;
+  }>();
+
+  const toggleLightTheme = () => {
+    $colorTheme = 'light';
+  };
+
+  const toggleDarkTheme = () => {
+    $colorTheme = 'dark';
+  };
+
+  $: {
+    if (browser) {
+      if ($colorTheme === 'light') {
+        document.documentElement.classList.remove('dark');
+      } else {
+        document.documentElement.classList.add('dark');
+      }
+    }
+  }
```

Comment:
Is there no general theme-switching logic that we should call into instead?

Reply:
Currently the theme button does the same logic: https://github.com/immich-app/immich/blob/ca9cad20bcec06bbb59b9a7c37f8cc16817bc36f/web/src/lib/components/shared-components/theme-button.svelte#L10-L18
Maybe we could move that logic to the preference store so that it happens automatically whenever `$colorTheme` updates